ui/src/lib/reactFlow/nodeTypes/GeneralNode/index.tsx

```diff
@@ -39,10 +39,10 @@ const GeneralNode: React.FC<GeneralNodeProps> = ({
 
   return (
     <div
-      className={`max-w-[200px] min-w-[150px] rounded-md border bg-secondary ${selected ? selectedColor : borderColor}`}>
+      className={`max-w-[200px] min-w-[150px] rounded-md border bg-secondary ${selected ? selectedColor : borderColor} ${data.isDisabled ? "opacity-70" : ""}`}>
       <div className="relative m-1 flex h-[25px] items-center gap-1 rounded-sm">
         <div
-          className={`flex justify-center self-center rounded-sm border p-1 align-middle ${selected ? selectedColor : borderColor} ${selected ? selectedBackgroundColor : className}`}>
+          className={`flex justify-center self-center rounded-sm border p-1 align-middle ${selected ? selectedColor : borderColor} ${selected ? selectedBackgroundColor : backgroundColor}`}>
```

Comment:
If you are adding the backgroundColor like this, you can update the individual nodes that are using this GeneralNode to not pass it anymore

Reply:
But also double check whether there are any other important classes now being missed due to you replacing className use